src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -426,6 +427,89 @@ public async Task SslStream_ClientCertificate_SendsChain()
             }
         }
 
+        [Theory]
+        [InlineData(16384 * 100, 4096, 1024, false)]
+        [InlineData(16384 * 100, 4096, 1024, true)]
+        [InlineData(16384 * 100, 1024 * 20, 1024, true)]
+        [InlineData(16384 , 3, 3, true)]
+        public async Task SslStream_RandomSizeWrites_OK(int bufferSize, int readBufferSize, int writeBufferSize, bool useAsync)
+        {
+            byte[] dataToCopy = RandomNumberGenerator.GetBytes(bufferSize);
+            byte[] dataReceived = new byte[dataToCopy.Length + readBufferSize]; // make the buffer bigger to have chance to read more
```

Comment:
Is `readBufferSize` here significant? Could we just do `dataToCopy.Length + 1`?

Reply:
somewhat. It makes `readBuffer.Slice(totalLength, readBufferSize)` to always work without worrying about how much is left.

Reply:
Ok, seems reasonable, but a comment would explain that :)